File: Solutions/34081/c++/program.cpp
// Copy from Quera

#include <iostream>

using namespace std ;

int main()
{
	int n,k;
	cin>>n>>k;
	int i=0 ;
	bool cmd=true ;
	int sum=0 ;
	while(true)
	{
		if(i==0 && cmd==false)
			break;
		cmd = false ;
		i+=k;
		if(i>=n)
			i-=n;
		sum++;
	}
	cout<<sum;
	return 0 ;
}

File: Solutions/34081/java/program.java
// Copy from Quera

import java.util.*;

class Main
{
	public static void main (String[] args)
	{
		Scanner input = new Scanner(System.in);
		int n = input.nextInt(), k = input.nextInt();
		int i=0 ;
    	boolean cmd=true;
    	int sum=0 ;
    	while(true)
    	{
    		if(i==0 && cmd==false)
    			break;
    		cmd = false ;
    		i+=k;
    		if(i>=n)
    			i-=n;
    		sum++;
    	}
		System.out.println(sum);
	}
}